src/Snapshotting/ConstructingAggregateRootRepositoryWithSnapshotting.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace EventSauce\EventSourcing\Snapshotting;
+
+use EventSauce\EventSourcing\AggregateRootId;
+use EventSauce\EventSourcing\AggregateRootRepository;
+use EventSauce\EventSourcing\Message;
+use EventSauce\EventSourcing\MessageRepository;
+use Generator;
+
+final class ConstructingAggregateRootRepositoryWithSnapshotting
+    implements AggregateRootRepositoryWithSnapshotting
+{
+    /**
+     * @var string
+     */
+    private $aggregateRootClassName;
+
+    /**
+     * @var MessageRepository
+     */
+    private $messageRepository;
+
+    /**
+     * @var SnapshotRepository
+     */
+    private $snapshotRepository;
+
+    /**
+     * @var AggregateRootRepository
+     */
+    private $regularRepository;
```

Comment:
Why not call it what it is? `$aggregateRootRepository`?

Reply:
Because this is the special one and the other one is regular.